oauth2client/service_account.py

```diff
@@ -108,6 +112,29 @@ def serialization_data(self):
             'private_key': self._private_key_pkcs8_text
         }
 
+    def to_json(self):
+        strip = (['_private_key'] + self.NON_SERIALIZED_MEMBERS)
```

Comment:
Instead, add a class attribute to `_ServiceAccountCredentials`

```python
NON_SERIALIZED_MEMBERS =  frozenset(
    ['_private_key'] +
    AssertionCredentials.NON_SERIALIZED_MEMBERS)
```

then you can just update `Credential.to_json` to use

```
self._to_json(self.NON_SERIALIZED_MEMBERS)
```

and just delete `OAuth2Credentials.to_json` (it is identical to `Credentials.to_json` anyhow).

We should also update `Credentials.NON_SERIALIZED_MEMBERS` to be a `frozenset`